ddtrace/tracer/tracer.go

```diff
@@ -171,9 +193,19 @@ func (t *tracer) worker() {
 			t.pushPayload(trace)
 
 		case <-ticker.C:
+			if t.statsd != nil {
+				tags := t.statsTags()
+				tags = append(tags, "reason:timeout")
```

Comment:
If I was observing metrics with "reason:timeout", that'd probably make me think something's going wrong and needs investigating.
And I wouldn't know what action / meaning to take for "payload-full".

Maybe they could be "scheduled:true" and "scheduled:false" or something to that effect.

Reply:
I agree. I'd say `reason:ticker` or `reason:size`, as in:

* Flushed because of size
* Flushed because of ticker